docs/RFCS/sql_privileges.md

```diff
@@ -0,0 +1,197 @@
+  A can remove `SELECT` from B. If another role granted `SELECT` as well, it stays).
+* databases share no data between them except for user tables. As such they are closer
+  to our databases then the postgres schema.
+* `SUPERUSER` is a user attribute. It allows everything.
+* postgres has a `PUBLIC` schema granting liberal privileges to all users.
+* privileges can be displayed using the `\du` command in psql.
+* `ALL` is translated to the list of all privileges.
+
+### Mysql
+
+* privileges are stored in a global table. It is thus possible to grant privileges
+  on objects that do not yet exist (eg: `GRANT ALL on db.* TO user1`).
+* privileges can be displayed using `SHOW GRANTS [FOR x]`.
+* `ALL` is a shortcut for setting and display.
+
+### Cockroach
+
+* the `root` user has default `ALL` privileges on all new databases. It is also the only
+  user allowd to create databases.
+* `root` user privileges cannot be removed. (This may change and be implicit `ALL` for `root`).
+* table privileges are inherited from database privileges at creation time. After that,
+  they are allowed to diverge.
+* `GRANT` is specified as a privilege. It may make more sense to use `WITH GRANT OPTION`
+  on the `GRANT` statement (similar to postgres and mysql).
+* privileges can be displayed using `SHOW GRANTS [FOR x]`.
+* `ALL` is a shortcut for setting and display.
+
+## Supported privileges
+
+| Privilege | Level     |
+|-----------|-----------|
+| ALL       | DB, Table |
+| CREATE    | DB        |
+| DROP      | DB, Table |
+| GRANT     | DB, Table |
+| SELECT    | Table     |
+| INSERT    | Table     |
+| DELETE    | Table     |
+| UPDATE    | Table     |
+
+## Statement comparison
+
+*Important note*: there is a lot of simplification in the following summaries.
+For more details, it is recommended to follow the links.
+
+`CREATE DATABASE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/create-database.html):
+	requires mysqladmin command or `CREATE` privilege on the database.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-createdatabase.html):
+	requires `SUPERUSER` or `CREATEDB` user option.
+* cockroach: requires `root` user.
+
+`CREATE TABLE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/create-table.html):
+	`CREATE` privilege on table.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-createtable.html):
+	`CREATE` privilege in schema.
+* cockroach: `CREATE` privilege on database.
+
+`DROP DATABASE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/drop-database.html):
+	`DROP` on database.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-dropdatabase.html):
+	database owner only.
+* cockroach: `DROP` on database, or `root` user.
+
+`DROP TABLE`:
+* [mysql](https://dev.mysql.com/doc/refman/5.7/en/drop-table.html):
+	`DROP` on table.
+* [postgres](http://www.postgresql.org/docs/9.4/static/sql-droptable.html):
+	table or schema owner. or `SUPERUSER`.
+* cockroach: `DROP` on table.
```

Comment:
nope. `root` gets `ALL` permissions by default, and they can't be taken off (that's negotiable though). This means that we still call `CheckPriv(root, DROP)` as opposed to `CheckPriv(user, DROP) || IsRoot()`